src/modal.ts

```diff
@@ -0,0 +1,113 @@
+export default class Modal {
```

Comment:
Same as Collapse, this should have a short explanation of the code, what it is doing and where it comes from